design.md

```diff
@@ -1,13 +1,47 @@
 # Design
 
+Each state is specified independently, with its rules in a block underneath the declaration. States are declared as
+
+
+```state StateName```
+
+
+where state names are arbitrary identifiers.
+Rules are declared with the surroundings followed by an arrow, then zero or more `go` instructions with a direction (`North`, ect.), zero or more `turn` instructions with either `Left`, `Right`, `Back`, or the cardinal directions, followed by an optional `transition` instruction with a new state. The first state declared in a program is the start state.
+
+
+Surroundings are represented by a list of one or more conditions, with the direction as a capital letter and specifiers "open" or "closed". Only one specifier per direction is allowed in any rule. If any directions are not specified, they will be assumed to be `*`. For example, `Nopen` is equivalent to `x***`, while `Wclosed, Sopen` is equivalent to `*x*W`. The directions can be specified in any order, and can either be cardinal ("NEWS") or relative ("FBLR" for "Forward, Back, Left, Right"). These two types of directions can't be mixed in the same state; for instance, "Fopen Eclosed" is not allowed.
```

Comment:
This is neat! Is it possible to extend the language to use shorter or longer syntaxes as well? Such as WO for Wopen and EC for Eclosed.